pkg/auth/providers/azure/azure_provider.go

```diff
@@ -415,23 +415,31 @@ func formAzureRedirectURL(config map[string]interface{}) string {
 	)
 }
 
-func extractAnnotations(config map[string]interface{}) map[string]string {
+// extractAnnotationsFromAuthConfig tries to extract the annotations from the AuthConfig value.
+// The AuthConfig value might come from either the database (on login attempts) or from the UI (on Azure AD setup attempts).
+// In these two cases, the structure of the config is different.
+// In the former, it's "metadata.annotations.[map of annotations]".
+// In the latter, it's "annotations.[map of annotations]". The function tries to find the annotations in either structure.
+func extractAnnotationsFromAuthConfig(config map[string]interface{}) map[string]string {
+	if metadata, ok := config["metadata"].(map[string]interface{}); ok {
+		return parseAnnotations(metadata)
+	}
+	logrus.Info("Failed to decode the 'metadata' field of the AuthConfig. Attempting to decode 'annotations' at the top level.")
+	return parseAnnotations(config)
+}
+
+func parseAnnotations(m map[string]interface{}) map[string]string {
```

Comment:
Would you mind changing the parameter here so that it is more clear what specific type of `m` you are expecting passed in? Something like `parseAnnotations(metadata map[string]interface{})` would work nicely.